chain/validation/types.go

```diff
@@ -0,0 +1,123 @@
+package validation
+
+import(
+	"fmt"
+	"github.com/pkg/errors"
+
+	"github.com/filecoin-project/chain-validation/pkg/state"
+	"github.com/filecoin-project/go-lotus/chain/address"
+	"github.com/filecoin-project/go-lotus/chain/types"
```

Comment:
When done correctly, this should not import anything from Lotus and it should be possible to move this into the chain-validator repo instead. It might be easier to actually do it there in the first place, so that broken constraints are immediately obvious.

Reply:
As this PR stands its too widely scoped. I'm going to close it out and refocus efforts on moving this code to the chain-validation package as you've suggested.